src/deploy/functions/params.ts

```diff
@@ -0,0 +1,332 @@
+import { logger } from "../../logger";
+import { FirebaseError } from "../../error";
+import { promptOnce } from "../../prompt";
+import * as build from "./build";
+import { assertExhaustive } from "../../functional";
+
+type CEL = build.Expression<string> | build.Expression<number> | build.Expression<boolean>;
+function isCEL(expr: string | number | boolean): expr is CEL {
+  return typeof expr === "string" && expr.includes("{{") && expr.includes("}}");
+}
+function dependenciesCEL(expr: CEL): string[] {
+  return /{{ params\.\w+ }}/.exec(expr)?.slice(1) || [];
```

Comment:
Aren't you missing a capture group here?

Reply:
You're right; and the really spooky thing is that the tests are passing on this commit. I'll figure out...whatever exec is returning here, and add a case.

Reply:
...which means this function has to go from `exec` to `match` also.